quickwit/quickwit-indexing/src/actors/doc_processor.rs

```diff
@@ -143,6 +177,83 @@ pub struct DocProcessor {
     timestamp_field_opt: Option<Field>,
     counters: DocProcessorCounters,
     publish_lock: PublishLock,
+    transform_layer: Option<VrlProgram>,
+}
+
+struct VrlProgram {
```

Comment:
Can we move the `VrlProgram` and related implementations out of `doc_processor`?